gcc: Depends on glibc if the host glibc is too old

sjackman opened the pull request:
Reverts the change made in
https://github.com/Homebrew/linuxbrew-core/commit/88a60db8de0dabdcb8ea5ee18b562f63b1df98e7#diff-baf58e557d0debf0bd219be97ec02ab8

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

When the host's `glibc` is too old, `brew install hello` ought to install both `glibc` and `gcc`.

sjackman commented:
This PR can be squash merged without updating the bottle.

maxim-belkin commented on Formula/gcc.rb:
```diff
@@ -40,6 +40,7 @@ class Gcc < Formula
   depends_on "mpfr"
   unless OS.mac?
     depends_on "binutils"
+    depends_on "glibc" if Formula["glibc"].installed? || OS::Linux::Glibc.system_version < Formula["glibc"].version
```
1. `.installed?` -> `.any_version_installed?`
2. I wonder if this `Formula["glibc"].*installed?` can cause problems with linkage:
- install gcc on a system with modern glibc
- install glibc
- run `brew linkage --test gcc`

sjackman replied:
https://github.com/Homebrew/linuxbrew-core/blob/59ff4310b038d86dfbaf3c8e784e560d06cda7f5/Formula/glibc.rb#L13

You cannot install brewed glibc on a system with a more up-to-date glibc.

sjackman replied:
> `.installed?` -> `.any_version_installed?`

👍 Now that I think about it though, this bit isn't needed. I've removed it.

iMichka commented:
I removed more code in that commit back then. Adding glibc back is not enough, you need to revive the stuff that was inside the `if build.with? "glibc"` condition.

And I think `if build.with?` is deprecated, this is why I removed this code.

sjackman commented:
> I removed more code in that commit back then. Adding glibc back is not enough, you need to revive the stuff that was inside the if build.with? "glibc" condition.
And I think if build.with? is deprecated, this is why I removed this code.

That code that was within `build.with? "glibc"` is not strictly necessary. This PR is sufficient as is.

sjackman commented:
@dawidd6 @iMichka Good to merge?

sjackman commented:
@dawidd6 Thanks for the ✅. Feel free to squash and merge at your leisure.
